src/mcp_server_docker/input_schemas.py

```diff
@@ -10,6 +10,35 @@
     field_validator,
     model_validator,
 )
+from pydantic.json_schema import GenerateJsonSchema, JsonSchemaValue
+from pydantic_core.core_schema import TupleSchema
+
+class CopilotAgentCompatibleSchema(GenerateJsonSchema):
+    """
+    Make **tuple fields** pass GitHub Copilot Agent’s tool-parameter validator.
+
+    Why?
+    ----
+    * Pydantic’s default tuple schema contains only ``prefixItems`` and **omits
+      ``items`` entirely**.
+    * Copilot Agent (v0.27.1) insists every array schema has an ``items`` entry
+      and fails with the exact message::
+
+          Failed to validate tool f1e_create_container: Error: tool parameters array type must have items
+
+    What we do
+    ----------
+    After generating the normal tuple schema, inject an empty object
+    ``items: {}`` if that key is missing.
+    An empty object means “any additional items are allowed”, which is valid
+    JSON Schema Draft 2020-12 *and* satisfies Copilot Agent.
+    """
+
+    def tuple_schema(self, schema: TupleSchema) -> JsonSchemaValue:
+        json_schema: JsonSchemaValue = super().tuple_schema(schema)
+        if json_schema.get("items") is None:
+            json_schema["items"] = {}
+        return json_schema
 
 
 class JSONParsingModel(BaseModel):
```

Comment:
Considering the name change above and the extension to `model_json_schema` could you also change the name of this model:

```suggestion
class McpServerDockerInputModel(BaseModel):
```